Pensel/firmware/modules/utilities/scheduler.c

```diff
@@ -0,0 +1,118 @@
+/*!
+ * @file    scheduler.c
+ * @author  Tyler Holmes
+ * @version 0.1.0
+ * @date    28-April-2018
+ * @brief   A rudamentary scheduler
+ */
+#include <stdint.h>
+#include <stdbool.h>
+
+#include "common.h"
+#include "scheduler.h"
+
+// TODO remove - debug
+// #include "peripherals/UART/UART.h"
+
+
+ret_t scheduler_init(schedule_t * schedule)
+{
+    schedule->num_slots_used = 0;
+    for (uint8_t i=0; i < NUM_SLOTS; i++) {
+        schedule->callbacks[i] = NULL;
+        schedule->callback_times_ms[i] = 0;
+        schedule->callback_last_run_ms[i] = 0;
+    }
+    return RET_OK;
+}
+
```

Comment:
In some of your source files you have 1 new line after functions, while in this one you have 2. Just a reminder that consistency is good ✌️

Reply:
Fact

Reply:
Maybe I should make a python linter...

Reply:
clang-format looks promising. Spinning this off into it's own branch for later